deepmd/dpmodel/descriptor/repflows.py

```
@@ -267,6 +273,16 @@ def __init__(
         self.edge_init_use_dist = edge_init_use_dist
         self.use_exp_switch = use_exp_switch
         self.use_dynamic_sel = use_dynamic_sel
+        # Snapshot the class-level backend choice on the instance.  This keeps
+        # descriptor/layer behavior stable even if a backend temporarily changes
+        # the class attribute while constructing a model.
+        self._use_static_dynamic_sel = type(self)._use_static_dynamic_sel
+        if self.use_dynamic_sel and self._use_static_dynamic_sel:
+            warnings.warn(
```

Ruff (0.15.17)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)